smoke-tests/src/test/groovy/io/opentelemetry/smoketest/LibertySmokeTest.groovy

```diff
@@ -0,0 +1,288 @@
+    true
+  }
+
+  def setupSpec() {
+    startTarget(jdk, serverVersion)
+  }
+
+  def cleanupSpec() {
+    stopTarget()
+  }
+
+  @Unroll
+  def "#appServer test static file found on JDK #jdk"(String appServer, String jdk) {
+    String url = "http://localhost:${target.getMappedPort(8080)}/app/hello.txt"
+    def request = new Request.Builder().url(url).get().build()
+    def currentAgentVersion = new JarFile(agentPath).getManifest().getMainAttributes().get(Attributes.Name.IMPLEMENTATION_VERSION)
+
+    when:
+    def response = CLIENT.newCall(request).execute()
+    TraceInspector traces = new TraceInspector(waitForTraces())
+    Set<String> traceIds = traces.traceIds
+    String responseBody = response.body().string()
+
+    then: "There is one trace"
+    traceIds.size() == 1
+
+    and: "Response contains Hello"
+    responseBody.contains("Hello")
+
+    and: "There is one server span"
+    traces.countSpansByKind(Span.SpanKind.SPAN_KIND_SERVER) == 1
+
+    and: "Expected span names"
+    traces.countSpansByName('/app/hello.txt') == 1
+
+    and: "The span for the initial web request"
+    traces.countFilteredAttributes("http.url", url) == 1
+
+    and: "Number of spans tagged with current otel library version"
+    traces.countFilteredResourceAttributes("telemetry.auto.version", currentAgentVersion) == 1
+
+    and:
+    traces.findResourceAttribute("os.name")
+      .map { it.stringValue }
+      .findAny()
+      .isPresent()
+
+    where:
+    [appServer, jdk] << getTestParams()
+  }
+
+  @Unroll
+  def "#appServer test static file not found on JDK #jdk"(String appServer, String jdk) {
+    String url = "http://localhost:${target.getMappedPort(8080)}/app/file-that-does-not-exist"
+    def request = new Request.Builder().url(url).get().build()
+    def currentAgentVersion = new JarFile(agentPath).getManifest().getMainAttributes().get(Attributes.Name.IMPLEMENTATION_VERSION)
+
+    when:
+    def response = CLIENT.newCall(request).execute()
+    TraceInspector traces = new TraceInspector(waitForTraces())
+    Set<String> traceIds = traces.traceIds
+
+    then: "There is one trace"
+    traceIds.size() == 1
+
+    and: "Response code is 404"
+    response.code() == 404
+
+    and: "There is one server span"
+    traces.countSpansByKind(Span.SpanKind.SPAN_KIND_SERVER) == 1
+
+    and: "Expected span names"
+    traces.countSpansByName('/app/file-that-does-not-exist') == 1
+
+    and: "The span for the initial web request"
+    traces.countFilteredAttributes("http.url", url) == 1
+
+    and: "Number of spans tagged with current otel library version"
+    traces.countFilteredResourceAttributes("telemetry.auto.version", currentAgentVersion) == 1
+
+    and:
+    traces.findResourceAttribute("os.name")
+      .map { it.stringValue }
+      .findAny()
+      .isPresent()
+
+    where:
+    [appServer, jdk] << getTestParams()
+  }
+
+  @Unroll
+  def "#appServer test request for WEB-INF/web.xml on JDK #jdk"(String appServer, String jdk) {
```

Comment:
Just out of curiosity: why this test is important here?

Reply:
Paths under WEB-INF are special, server should not show content for them. At least liberty cuts off request processing rather early for WEB-INF/web.xml. Added a test to verify that we do get a span for such a request.